src/plugins/util/OverlayPlugin.js

```diff
@@ -300,6 +305,27 @@ class OverlayPlugin {
      * --------------------------
      */
 
+    // CheckModalPositionOutsideVideoWrapper generated by ai
+    /**
+     * Checks if modal is within video wrapper bounds
+     * @returns {boolean} True if modal is outside video wrapper bounds, false otherwise
+     */
+    checkModalPositionOutsideVideoWrapper() {
+        const videoWrapperRect = this.instance.videoWrapper.getBoundingClientRect();
+        const modalRect = this.widget.getBoundingClientRect();
+
+        // Check if modal is outside video wrapper bounds
+        if (
+            modalRect.right > videoWrapperRect.right ||
+            modalRect.left < videoWrapperRect.left ||
+            modalRect.bottom > videoWrapperRect.bottom ||
+            modalRect.top < videoWrapperRect.top
+        ) {
+            return true;
+        }
+        return false;
```

Comment:
Could be simpler, no? Something like this (not proper indentation):
```javascript
return modalRect.right > videoWrapperRect.right ||
            modalRect.left < videoWrapperRect.left ||
            modalRect.bottom > videoWrapperRect.bottom ||
            modalRect.top < videoWrapperRect.top
```

Reply:
As discussed, this is subjective. Feel free to keep the code you find most readable!